modules/integration/tests-integration/bpmn/src/test/java/org/wso2/bps/integration/tests/bpmn/JavaServiceTaskDeploymentUndeploymentTestCase.java

```diff
@@ -0,0 +1,104 @@
+        File javaArtifact = new File(artifactLocation);
+        Loader.copyToComponentLib(javaArtifact);
+        Loader.restartForcefully();
+
+        //reintialising as session cookies and other configuration which expired during restart is reset
+        init();
+    }
+
+
+    @Test(groups = {"wso2.bps.test.deploy.JavaServiceTask"}, description = "Deploy/UnDeploy Package Test", priority = 1, singleThreaded = true)
+
+    public void deployUnDeployJavaServiceTaskBPMNPackage() throws Exception {
+        init();
+        ActivitiRestClient tester = new ActivitiRestClient(bpsServer.getInstance().getPorts().get("http")
+                , bpsServer.getInstance().getHosts().get("default"));
+
+        String filePath = FrameworkPathUtil.getSystemResourceLocation() + File.separator
+                          + BPMNTestConstants.DIR_ARTIFACTS + File.separator
+                          + BPMNTestConstants.DIR_BPMN + File.separator + "sampleJavaServiceTask.bar";
+
+        String fileName = "sampleJavaServiceTask.bar";
+        String[] deploymentResponse;
+
+        deploymentResponse = tester.deployBPMNPackage(filePath, fileName);
+        Assert.assertTrue("Deployment Successful", deploymentResponse[0].contains(BPMNTestConstants.CREATED));
+        String[] deploymentCheckResponse = tester.getDeploymentInfoById(deploymentResponse[1]);
+        Assert.assertTrue("Deployment Present", deploymentCheckResponse[2].contains(fileName));
+
+
+        String[] definitionResponse = tester.findProcessDefinitionInfoById(deploymentResponse[1]);
+        Assert.assertTrue("Search Success", definitionResponse[0].contains("200"));
+
+        //Starting and Verifying Process Instance
+        String[] processInstanceResponse = tester.startProcessInstanceByDefintionID(definitionResponse[1]);
+        Assert.assertTrue("Process Instance Started", processInstanceResponse[0].contains(BPMNTestConstants.CREATED));
+        String searchResponse = tester.searchProcessInstanceByDefintionID(definitionResponse[1]);
+        Assert.assertTrue("Process Instance Present", searchResponse.contains(BPMNTestConstants.OK));
+
+        //check for varible value if true
+        String[] innovationResponse = tester.getValueOfVariableOfProcessInstanceById(processInstanceResponse[1], "executionState");
+        Assert.assertTrue("Variable Present", innovationResponse[0].contains(BPMNTestConstants.OK));
+        Assert.assertTrue("Variable Present", innovationResponse[1].contains("executionState"));
+        Assert.assertTrue("Variable Value is True", innovationResponse[2].contains("true"));
+
+
+        //Suspending the Process Instance
+        String[] suspendResponse = tester.suspendProcessInstanceById(processInstanceResponse[1]);
+        Assert.assertTrue("Process Instance has been suspended", suspendResponse[0].contains(BPMNTestConstants.OK));
+        Assert.assertTrue("Process Instance has been suspended", suspendResponse[1].contains("true"));
```

Comment:
There should be methods to get suspended/active process instances. Invoke those and verify whether the process instance is moved to suspended state, in addition to checking return values.